src/app.ts

```diff
@@ -71,6 +76,64 @@ const main = async () => {
     res.json({ success: 'true' });
   });
 
+  const storage = multer.memoryStorage();
+  const upload = multer({ storage: storage });
+
+  // Not really sure where in the code base go, but here's the working route!
+  // I did my part :)
+  // Is just using `/flyers/` the right route here?
+  app.post('/flyers/', upload.single('file'), async (req, res) => {
+    // Ensure there is an image file present
+    if (!req.file) {
+      return res.status(400).json({ error: 'No file uploaded' });
+    }
+    const {
+      categorySlug,
+      endDate,
+      flyerURL,
+      location,
+      organizationID,
+      startDate,
+      title,
+    } = req.body;
+
+    // Is there a better way to do this? (assert request body has required fields)
+    if (
+      categorySlug == null ||
+      endDate == null ||
+      flyerURL == null ||
+      location == null ||
+      organizationID == null ||
+      startDate == null ||
+      title == null
+    ) {
+      return res.status(400).json({ error: 'Missing a required field' });
+    }
+
+    // Get the file from form-data and await the upload service
+    const imageB64 = req.file.buffer.toString('base64');
+    const imageURL = await utils.uploadImage(imageB64);
```

Comment:
Have you checked that the upload service can handle a massive Base64 string sent via a network request (i.e., have you tried uploading a really large file and seeing if that works)

Reply:
I checked it just now, and it crashes the backend as utils throws an unchecked exception. Not really sure what to do considering the upload service is sort of a black box since we're using AppDev's upload service and not s3 or something. Do we need to rewrite AppDev's upload service to use form-data too? I feel like there's not much I can change on Volume's backend to fix this other than wrapping in a try catch.

Reply:
Yeah I spoke about this with Vin but we would have to add an endpoint to the upload service that can handle a form request, and then you can just forward req.file as part of a form request to the endpoint in the upload service

Reply:
right, but I can't do that right now since the upload service can't handle a form right now, so is this ok for right now?

Reply:
The issue we're trying to solve is that certain orgs can't upload files because the files are too large right? So if we can't deal with large files then we would still have this issue even if we merge this code right? I think if we want to resolve the file size issue once and for all we would have to make that new endpoint in the upload service (which should hopefully be just a couple lines of code because I think `boto3` which is used in the upload service has support for this).

Reply:
I agree for sure, I just wasn't sure if I had access to change the upload service. I found the GitHub repository and started working on a fix. I will also update this PR again to use form data in the post request to the upload service once I am done.